core/src/test/scala/quasar/plugin/postgres/PostgresDestinationSpec.scala

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright 2014–2019 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.plugin.postgres
+
+import slamdata.Predef.{Stream => _, _}
+
+import argonaut._, Argonaut._, ArgonautScalaz._
+
+import cats.effect._
+import cats.implicits._
+
+import doobie._
+import doobie.implicits._
+import doobie.util.Read
+
+import fs2.{Pull, Stream}
+
+import java.net.URI
+import java.time._
+
+import org.specs2.matcher.MatchResult
+
+import qdata.time._
+
+import quasar.EffectfulQSpec
+import quasar.connector._
+import quasar.api.destination._
+import quasar.api.resource._
+import quasar.api.table.{ColumnType, TableColumn}
+import quasar.contrib.scalaz.MonadError_
+
+import scala.Float
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import scalaz.-\/
+import scalaz.syntax.show._
+
+import shapeless._
+import shapeless.ops.hlist.{Mapper, ToList}
+import shapeless.ops.record.{Keys, Values}
+import shapeless.record._
+import shapeless.syntax.singleton._
+
+import shims.applicativeToScalaz
+
+object PostgresDestinationSpec extends EffectfulQSpec[IO] with CsvSupport with PGTemporalMeta {
```

Comment:
Btw is there a nice script or something we can use to set up and tear down the docker stuff for the testing? If it helps, fs2-ssh has some Docker utilities which keeps everything confined within the Specs2 framework, which is kind of nice.

Reply:
@djspiewak From the root of the repo `docker-compose up postgres{9,10,11}`

Reply:
Kk